ml/src/main/scala/ai/h2o/sparkling/ml/params/H2OGridSearchParams.scala

```diff
@@ -102,7 +102,50 @@ trait H2OGridSearchParams extends H2OCommonSupervisedParams {
 
   def getSeed(): Long = $(seed)
 
-  def getNfolds(): Int = $(nfolds)
+  @DeprecatedMethod("the 'getPredictionCol' method of a given algorithm", "3.32")
+  def getPredictionCol(): String = getAlgo().getPredictionCol()
```

Comment:
What if the user does:

`grid.setLabelCol("X).setAlgo(new H2OGBM())`. Will label col value be preserved?

Reply:
the `propagateValuesToAlgorithm` dictionary serves for that purpose. Unfortunately, I wasn't able to test it becouse of https://0xdata.atlassian.net/browse/SW-2276

Reply:
I'm probably missing something -> I can see `propagateValuesToAlgorithm` in Python implementation, but missing it in Scala

Reply:
@jakubhava I've tried to address this comment, please can you take a look?

Reply:
Thank you, looks good